recipes/leptonica/all/conanfile.py

```diff
@@ -101,6 +123,9 @@ def generate(self):
         if Version(self.version) >= "1.83.0":
             tc.variables["LIBWEBP_SUPPORT"] = self.options.with_webp
             tc.variables["OPENJPEG_SUPPORT"] = self.options.with_openjpeg
+        if Version(self.version) >= "1.84.0" and self.options.with_openjpeg:
```

Comment:
I see the version 1.84.0 added options to enable dependencies: https://github.com/DanBloomberg/leptonica/compare/1.83.1...1.84.1#diff-1e7de1ae2d059d21e1dd75d5812d5a34b0222cef273b7c3a2af62eb747f9d20aR65

It could be not only added to this recipe, but also fix this preprocessor workaround.